source/opt/instruction.cpp

```diff
@@ -369,5 +374,32 @@ Instruction* Instruction::InsertBefore(std::unique_ptr<Instruction>&& i) {
   i.get()->InsertBefore(this);
   return i.release();
 }
+
+bool Instruction::IsValidBasePointer(Instruction* inst) const {
+  if (context()->module()->HasCapability(SpvCapabilityAddresses)) {
```

Comment:
This is really permissive.   Suggest updating the method comment saying we assume inst is an instruction that yields a pointer.

Or move move the check for pointer type up above this line.

Reply:
I've moved the check for pointer type.  I've also the check for the opcodes of non variables.  For those I don't case so much about the opcodes.  I care about the type.